src/jdk.incubator.jpackage/linux/classes/jdk/incubator/jpackage/internal/DesktopIntegration.java

```diff
@@ -524,6 +546,7 @@ private static String stringifyShellCommands(List<String> commands) {
 
     private final List<LinuxFileAssociation> associations;
 
+    private static boolean initAppImageLaunchers = true;
```

Comment:
What is the logic behind static variable?

Reply:
It is used to avoid infinite calls in DesktopIntegration(). I modified it by removing PREDEFINED_APP_IMAGE from launcherParams.